File: app/src/main/java/com/example/mhmdreza_j/xproject/views/game/FinishGameFragment.java
package com.example.mhmdreza_j.xproject.views.game;


import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.Toast;

import com.example.mhmdreza_j.xproject.R;
import com.example.mhmdreza_j.xproject.lib.blurry.Blurry;
import com.example.mhmdreza_j.xproject.utils.BlurryUtil;
import com.example.mhmdreza_j.xproject.views.base_class.BaseFragment;
import com.example.mhmdreza_j.xproject.views.main_page.MainActivity;
import com.example.mhmdreza_j.xproject.views.main_page.MainFragment;
import com.facebook.FacebookActivity;

import java.util.ArrayList;
import java.util.concurrent.LinkedBlockingDeque;

import static com.example.mhmdreza_j.xproject.utils.UIUtils.dp;
import static com.example.mhmdreza_j.xproject.views.game.QuestionFragment.Mode.GREEN;
import static com.example.mhmdreza_j.xproject.views.game.QuestionFragment.Mode.YELLOW;

/**
 * A simple {@link Fragment} subclass.
 */
public class FinishGameFragment extends BaseFragment {


    private ArrayList<View> views = new ArrayList<>();

    public FinishGameFragment() {
        // Required empty public constructor
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        final View view = inflater.inflate(R.layout.fragment_finish_game, container, false);
        final LinearLayout questionLayout = view.findViewById(R.id.questionLayout);
        if (views.size() == 0){
            for (int i = 0; i < 10; i++) {
                final View v = new View(getContext());
                v.setBackgroundResource(i % 2 == 0 ? R.drawable.question_yellow : R.drawable.question_green);
                views.add(v);
                final int finalI = i;
                v.setOnClickListener(new View.OnClickListener() {
                    @Override
                    public void onClick(View v1) {
                        if (getActivity() == null) {
                            return;
                        }
                        Blurry.ImageComposer composer = Blurry.with(v1.getContext())
                                .color(R.color.dark)
                                .capture(view);
                        BlurryUtil.saveQuestionComposer(composer);
                        QuestionFragment fragment = new QuestionFragment();
                        fragment.setLeft(v1.getLeft() + questionLayout.getLeft());
                        fragment.setTop(v1.getTop()+ questionLayout.getTop());
                        fragment.setMode(finalI % 2 == 0 ? YELLOW: GREEN);

                        ((MainActivity) getActivity()).startFragment(fragment);
                    }
                });
                questionLayout.addView(v, dp(30 , getContext()), dp(30, getContext()));
            }
        }

        ImageView homeButton = view.findViewById(R.id.homeButton);
        homeButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                onBackPressed();
            }
        });
        return view;
    }

    @Override
    public void onBackPressed() {
        if (getActivity() == null) return;
        ((MainActivity) getActivity()).startFragment(new MainFragment());
    }
}

File: app/src/main/java/com/example/mhmdreza_j/xproject/views/game/FinishGameFragment.kt
package com.example.mhmdreza_j.xproject.views.game


import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.View.VISIBLE
import android.view.ViewGroup
import com.example.mhmdreza_j.xproject.R
import com.example.mhmdreza_j.xproject.lib.blurry.Blurry
import com.example.mhmdreza_j.xproject.utils.BlurryUtil
import com.example.mhmdreza_j.xproject.utils.FINISH_GAME_KEY
import com.example.mhmdreza_j.xproject.utils.dp
import com.example.mhmdreza_j.xproject.utils.setProfileImage
import com.example.mhmdreza_j.xproject.views.base_class.BaseFragment
import com.example.mhmdreza_j.xproject.views.game.socket_model.GameState
import com.example.mhmdreza_j.xproject.views.game.socket_model.QuestionModel
import com.example.mhmdreza_j.xproject.views.main_page.MainFragment
import ir.tapsell.sdk.bannerads.TapsellBannerType
import kotlinx.android.synthetic.main.fragment_finish_game.*
import kotlin.math.min

const val MY_ANSWER = "MY_ANSWER"
const val CORRECT_ANSWER = "CORRECT_ANSWER"
const val QUESTION = "QUESTION"
const val TOP = "TOP"
const val LEFT = "LEFT"

class FinishGameFragment : BaseFragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_finish_game, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        bannerView.loadAd(activity, FINISH_GAME_KEY, TapsellBannerType.BANNER_300x250)
        initViews()
    }

    private fun initViews() {
        addViews()
        homeButton.setOnClickListener { onBackPressed() }

        setProfileImage(profileImageView, arguments?.getInt(MY_AVATAR))
        setProfileImage(opponentImageView, arguments?.getInt(OPPONENT_AVATAR))

        when (arguments?.getSerializable(GAME_STATE)) {
            GameState.WIN -> myWinImage.visibility = VISIBLE
            GameState.LOSE -> opponentWinImage.visibility = VISIBLE
            else -> toastMessage("بازی مساوی شد!")
        }
    }

    private fun addViews() {
        val answers = arguments?.getSerializable(ANSWERS) as ArrayList<*>
        val myAnswers = arguments?.getSerializable(MY_ANSWERS) as ArrayList<String>
        val questions = arguments?.getSerializable(QUESTIONS) as ArrayList<*>
        normalizeMyAnswer(myAnswers, questions)
        val index = min(questions.size, myAnswers.size)
        for (i in 0..4) {
            val v = View(context)
            val inBoundIndex = i < index
            val isCorrectAnswer = if (inBoundIndex) answers[i] == myAnswers[i] else false
            v.setBackgroundResource(when {
                i >= questions.size -> R.drawable.question_transparent // not shown question
                isCorrectAnswer -> R.drawable.question_green // wrong answered question
                else -> R.drawable.question_red// right answered question
            })
            v.setOnClickListener(View.OnClickListener { v1 ->
                if (activity == null || i >= questions.size) {
                    return@OnClickListener
                }
                val composer = Blurry.with(v1.context)
                        .color(R.color.dark)
                        .capture(view)
                BlurryUtil.saveQuestionComposer(composer)
                val fragment = QuestionFragment()
                val bundle = Bundle()
                bundle.putInt(LEFT, v1.left + questionLayout.left)
                bundle.putInt(TOP, v1.top + questionLayout.top)
                bundle.putSerializable(QUESTION, questions[i] as QuestionModel)
                bundle.putString(CORRECT_ANSWER, answers[i] as String)
                bundle.putString(MY_ANSWER, myAnswers[i])
                fragment.arguments = bundle
                mainActivity?.startFragment(fragment)
            })
            questionLayout.addView(v, dp(40f, questionLayout.context), dp(40f, questionLayout.context))
        }
    }

    private fun normalizeMyAnswer(myAnswers: ArrayList<String>, questions: ArrayList<*>) {
        while (myAnswers.size < questions.size) {
            myAnswers.add("")
        }
    }

    override fun onBackPressed() {
        if (activity == null) return
        mainActivity?.startBackgroundMusic()
        mainActivity!!.startFragment(MainFragment())
    }
}
